modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java

```diff
@@ -245,6 +250,14 @@ public boolean isShow(HttpServletRequest httpServletRequest)
 	protected void activate() {
 		_portletNamespace = _portal.getPortletNamespace(
 			AnalyticsReportsPortletKeys.ANALYTICS_REPORTS);
+
+		CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+
+		cacheBuilder.expireAfterAccess(10, TimeUnit.MINUTES);
```

Comment:
@marcosapmf Currently the values in this cache will only be invalided due to this 10-minute expiration. So, if the permission gets changed, the change will only take effect after this cache proactively expire the previous result. If there's no proper way to invalidate cache values upon permission changes, please do not cache it.